static/js/render_markers.js

```diff
@@ -1,4 +1,105 @@
+
+function drawMarkerInfoBox(acc_num){
+  var accNumInfo = acc_num + " accident(s) showing in screen";
+  var colorInfo = "Color: severity K, <span style='color:red'>A</span>, "+
+    "<span style='color:orange;'>B</span>, "+
+    "<span style='color:green'>C</span>";
+  var iconInfo = "<img src='/static/imgs/collision/head_on_black_50p.png' /> Head on<br/>"+
+    "<img src='/static/imgs/collision/rear_end_black_50p.png' />Rear end <br/>"+
+    "<img src='/static/imgs/collision/pedestrian_black_50p.png' />Pedestrian<br/>"+
+    "<img src='/static/imgs/collision/bicycle_black_50p.png' />Bicycle<br/>"+
+    "<img src='/static/imgs/collision/dear_black_50p.png' />Animal";
+  
+  var whoInfo = accNumInfo+"<br/>"+colorInfo+"<br/>"+iconInfo;
+  $('#info-box').html(whoInfo);
+}
+
+function getAccidentMarkerCB(map, accidents) {
+  /* main entrance to render the markers
+   */
+  clearVisual();
+  set_markers(homeJS.onscreenMarker, null);
+  homeJS.onscreenMarker = getAndRenderMarkers(map, accidents);
+  drawMarkerInfoBox(  homeJS.onscreenMarker.length );
+
+  // TODO, why set the visual mode here? should be controled by human?
+  homeJS.currentVisualMode = 'markers';
+  
+  console.log("rendering the accident marker");
+}
+
+function set_markers(markers, map){
+  /* add the markers to map
+   */
+  for(var i = 0; i < markers.length; ++i) {
+    markers[i].setMap(map);
+  }
+}
+
+function getAccidentIcon(accident){
+  /* severity K,A,B,C correspond to color (black, red, orange, green)
+     the icon will change based on the accident collsion type
+   */
+  // the severity only for the worst case
+  var severityLevel = 'green';
+  if(accident.num_b>0) severityLevel = 'orange';
+  if(accident.num_a>0) severityLevel = 'red';
+  if(accident.num_k>0) severityLevel = 'black';
+
+  // collision type priority
+  // pedestrian 14
+  // cyclist 15
+  // head-on 27
+  // rear end 21, 22
+  //console.log(accident.events);
+  var collision_type = '';
+  if( accident.events.length > 1){
+    if( (21 in accident.events) || (22 in accident.events))

```

Comment:
This may not the a good way to write it. The magic number here may cause confusion.